src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/ConsiderPassingBaseTypesAsParameters.cs

```diff
@@ -0,0 +1,225 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Analyzer.Utilities.PooledObjects;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.CodeQuality.Analyzers.ApiDesignGuidelines
+{
+    using static MicrosoftCodeQualityAnalyzersResources;
+
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public sealed class ConsiderPassingBaseTypesAsParameters : DiagnosticAnalyzer
+    {
+        internal const string RuleId = "CA1011";
+
+        internal static readonly DiagnosticDescriptor Rule =
+            DiagnosticDescriptorHelper.Create(
+                RuleId,
+                CreateLocalizableResourceString(nameof(ConsiderPassingBaseTypesAsParametersTitle)),
+                CreateLocalizableResourceString(nameof(ConsiderPassingBaseTypesAsParametersMessage)),
+                DiagnosticCategory.Design,
+                RuleLevel.IdeSuggestion,
```

Comment:
According to https://github.com/dotnet/runtime/issues/61916#issuecomment-977195070, this should be Hidden, not an IDE suggestion.

Reply:
Thanks!

Reply:
Yes, we proposed to be hidden with that comment, but it is approved with IdeSuggestion https://github.com/dotnet/runtime/issues/61916#issuecomment-982973839, no need to change